### PR TITLE
Refactor to exclude folders from installing packages

### PR DESCRIPTION
**Pull request by toha87**
Exclude specific folders from installing packages

Environment variable called "EXCLUDE_FOLDERS" with an array of strings. Note: case-sensitive
       Ex: packages, containers

Downside: if multiple folders have the same name all of them will be ignored

### CODE REVIEW

File: recursive-install.js [hunk 1]
```diff
@@ -5,58 +5,101 @@ var shell = require('shelljs')
...
+            const foldersToExclude = excludeFolders
+                .split(",")
+                .map((str) => {
+                    /**
+                     * Remove white space from the beginning or end of a string
+                     */
+                    if (typeof str === 'string') return str.trim();
+                });
+
+            if(foldersToExclude && Array.isArray(foldersToExclude) && foldersToExclude.length > 0) {
```
Comment (makrasnov100) [inline recursive-install.js]: Array check also does null check
```suggestion
            if(Array.isArray(foldersToExclude) && foldersToExclude.length > 0) {
```

File: recursive-install.js [hunk 2]
```diff
@@ -5,58 +5,101 @@ var shell = require('shelljs')
...
-      execSync('npm install --production', { cwd: dir})
-    } else {
-      console.log('Installing ' + dir + '/package.json')
-      execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', { cwd: dir})
-      execSync('npm install', { cwd: dir})
+function getPackageJsonLocations(dirname) {
+    return shell.find(dirname)
+        .filter(function (fname) {
+            return !(fname.indexOf('node_modules') > -1 || fname[0] === '.') &&
+                path.basename(fname) === 'package.json'
```
Comment (makrasnov100) [inline recursive-install.js:16]: could be more explicit with variable name `fname` -> `filename` for readability

File: recursive-install.js [hunk 3]
```diff
@@ -5,58 +5,101 @@ var shell = require('shelljs')
...
+
+        if (argv.production) {
+            console.log('Installing ' + dir + '/package.json with --production option')
+            execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', {cwd: dir})
+            execSync('npm install --production', {cwd: dir})
+        } else {
+            console.log('Installing ' + dir + '/package.json')
+            execSync('echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc', {cwd: dir})
+            execSync('npm install', {cwd: dir})
+        }
```
Comment (makrasnov100) [inline recursive-install.js:75]: seems you could use `argv && argv.production === true ? "--production" : ""` to avoid some code duplication

**Review by makrasnov100 — APPROVED** [review]
Looks great, some tiny things below. Also I feel like the main file could use webstorm's auto formatting and 2 space indent so everything looks more like our codebase.

File: README.md [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 ===
 
 A small utility to recursively run `npm install` in any child directory that has a `package.json` file excluding sub directories of `node_modules`.
 
 Preinstall
 ---
-You will need an `NPM_TOKEN` environment variable, and it should contain your npm `Auth Token`
+You will need an `NPM_TOKEN` environment variable, and it should contain your npm `Auth Token`.
+Optionally include `EXCLUDE_FOLDERS` environment variable to exclude folders
+Ex: `packages, cypress` Note: case-sensitive
```
Comment (blazinaj) [inline README.md:10]: Thank you for the docs :)

**Review by blazinaj — APPROVED** [review]
This looks great :) thank you

**Pull request merged**